skimage/morphology/_flood_fill.py

```diff
@@ -0,0 +1,290 @@
+        considered as part of the neighborhood (fully connected).
+    connectivity : int, optional
+        A number used to determine the neighborhood of each evaluated pixel.
+        Adjacent pixels whose squared distance from the center is larger or
+        equal to `connectivity` are considered neighbors. Ignored if
+        `selem` is not None.
+    tolerance : float or int, optional
+        If None (default), adjacent values must be strictly equal to the
+        initial value of `image` at `seed_point`.  This is fastest.  If a value
+        is given, a comparison will be done at every point and if within
+        tolerance of the initial value will also be filled (inclusive).
+
+    Returns
+    -------
+    mask : ndarray
+        A Boolean array with the same shape as `image` is returned, with values
+        equal to 1 for areas connected to and equal (or within tolerance of)
+        the seed point.  All other values are zero.
+
+    Notes
+    -----
+    The conceptual analogy of this operation is the 'paint bucket' tool in many
+    raster graphics programs.  This function returns just the mask
+    representing the fill.
+
+    If indices are desired rather than masks for memory reasons, the user can
+    simply run `numpy.nonzero` on the result, save the indices, and discard
+    this mask.
+
+    Examples
+    --------
+    >>> from skimage.morphology import flood
+    >>> image = np.zeros((4, 7), dtype=int)
+    >>> image[1:3, 1:3] = 1
+    >>> image[3, 0] = 1
+    >>> image[1:3, 4:6] = 2
+    >>> image[3, 6] = 3
+    >>> image
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [1, 0, 0, 0, 0, 0, 3]])
+
+    Fill connected ones with 5, with full connectivity (diagonals included):
+
+    >>> mask = flood(image, (1, 1))
+    >>> image[mask] = 5
+    >>> image
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [0, 5, 5, 0, 2, 2, 0],
+           [5, 0, 0, 0, 0, 0, 3]])
+
+    Fill connected fives with 1, with only cardinal direction connectivity:
+
+    >>> mask = flood(image, (1, 1), connectivity=1)
+    >>> image[mask] = 1
+    >>> image
+    array([[0, 0, 0, 0, 0, 0, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [0, 1, 1, 0, 2, 2, 0],
+           [5, 0, 0, 0, 0, 0, 3]])
+
+    Fill with a tolerance:
+
+    >>> mask = flood(image, (0, 0), tolerance=1)
+    >>> image[mask] = 5
+    >>> image
+    array([[5, 5, 5, 5, 5, 5, 5],
+           [5, 5, 5, 5, 2, 2, 5],
+           [5, 5, 5, 5, 2, 2, 5],
+           [5, 5, 5, 5, 5, 5, 3]])
+    """
+    # Correct start point in ravelled image - only copy if non-contiguous
+    image = np.asarray(image)
+    if image.flags.f_contiguous is True:
+        order = 'F'
+    elif image.flags.c_contiguous is True:
+        order = 'C'
+    else:
+        warnings.warn('Non-contiguous array passed as `image`; this will be '
+                      'converted to a contiguous array as a copy.')
```

Comment:
I really like how explicit this!  But you may want to add `stacklevel=3` here to raise the warning on the stack level of the user code instead. Same for the warning in `flood_fill`.

Reply:
3? Not 2? But anyway I think this is probably the right place for the warning...

Reply:
No, you are right! I got confused because I remembered https://github.com/scikit-image/scikit-image/issues/2097#issuecomment-428118113. But that one uses the wrapped `warn` function unlike this case. So it should be 2.

Reply:
Given that it's possible to fill inplace even with a noncontiguous array, I'm going to remove all of these warnings.  The original won't be made contiguous and will be filled, so I don't see why we need them.